src/components/DealForm.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+
+import useDealForm from '../hooks/useDealForm';
+
+import TextField from './forms/TextField';
+import TextAreaField from './forms/TextAreaField';
+import CheckBoxField from './forms/CheckBoxField';
+import Button from './Button';
+
+const Form = styled.form`
+  width: 100%;
+  display: grid;
+  grid-gap: 2rem;
+  max-width: 100%;
+  grid-template-columns: 50%;
+  grid-template-rows: 6rem 6rem auto auto 6rem 6rem;
+
+
+`;
+
+
+const Submit = styled(Button)`
+  grid-column: 1 / end;
+  justify-self: center;
+  align-self: center;
+  width: 18rem;
+  font-size: 1.8rem;
+  margin-left: auto;
+
+`;
+function DealForm({ headerComponent: HeaderComponent, onSubmitComplete }) {
+  const onSuccess = () => {
+    console.log('success');
```

Comment:
Do we need the console.log here?

Reply:
Sorry, must have left it during debugging 😄